pkg/k8sutils/job.go

```diff
@@ -258,13 +263,41 @@ func (k8s *k8sImpl) prepareJobEnv(task config.Task, eventData *keptnv2.EventData
 	return jobEnv, nil
 }
 
-func generateEnvFromEvent(env config.Env, jsonEventData interface{}) ([]v1.EnvVar, error) {
+func generateEnvFromEvent(env config.Env, jsonEventData interface{}, formatting string) ([]v1.EnvVar, error) {
 
 	value, err := jsonpath.Get(env.Value, jsonEventData)
 	if err != nil {
 		return nil, fmt.Errorf("could not add env with name %v, value %v, valueFrom %v: %v", env.Name, env.Value, env.ValueFrom, err)
 	}
 
+	if strings.EqualFold(formatting, "json") {
+		jsonString, err := json.Marshal(value)
+
+		if err != nil {
+			log.Printf("Error converting environment variable value to JSON: %e",err)
+		}
+
+		value = string(jsonString[:])
+	} else if strings.EqualFold(formatting, "yaml") {
+		yamlString, err := yaml.Marshal(value)
+
+		if err != nil {
+			log.Printf("Error converting environment variable value to YAML: %e",err)
+		}
+
+		value = string(yamlString[:])
+	} else if reflect.ValueOf(value).Kind() == reflect.Map {
+		log.Println("Values is map")
+
+		jsonString, err := json.Marshal(value)
```

Comment:
Ok, so your solution is implementing option1+2 in #59. Its fine by me, since we don't parse the map putput. However, this is a breaking change which needs to be documented.

Reply:
Documentation has been added. Should the breaking change also be mentioned in the upcoming release?

Reply:
Cool, thanks @TannerGabriel! :rocket: Yes, makes sense. @yeahservice can you please add a note to #62?

Reply:
Will do